entity-service-change-event-generator/src/main/java/org/hypertrace/entity/service/change/event/impl/EntityChangeEventGeneratorImpl.java

```diff
@@ -147,6 +166,33 @@ private void sendUpdateNotification(
     }
   }
 
+  private boolean shouldSendNotification(
+      RequestContext requestContext, Entity prevEntity, Entity currEntity) {
+    String entityType = prevEntity.getEntityType();
+    Entity.Builder prevEntityBuilder = prevEntity.toBuilder();
+    Entity.Builder currEntityBuilder = currEntity.toBuilder();
+    this.changeNotificationSkipAttributeList.forEach(
+        attributeId -> {
+          if (!attributeId.startsWith(entityType)) {
```

Comment:
This is one way to define whether given attribute matches the entity type or not. The other way could be to define the config as an map of entity types vs attributeID. something similar we had before like this
```
{
    scope = API
    attributeIds = ["last_called_millis"]
  }
```
@aaron-steinfeld thoughts?

Reply:
> {
    scope = API
    attributeIds = ["last_called_millis"]
  }

could be just the example (that's a subdoc path), but that's the whole thing we're trying to get away from. Just to clarify terminology:
```
scope: API
attribute key: lastCalledTimestamp
attribute ID: API.lastCalledTimestamp
subDocPath: last_called_millis (or maybe it was attributes.last_called_millis, can't remember)
```

Now when we get an entity object, we've got a scope and a subdocpath, and we want to get that to the common platform attribute model - either a scope + attr key, or attribute ID.

We know how to get the attribute ID for explicitly mapped ones from the other config used in EntityAttributeMapping. We also know how to reverse the logic for the implicitly mapped ones to get to an attribute key (and already have scope). So in one path we end up with an attribute ID, in the other path an attribute key. Either one is sufficient for config, but ideally we want them to be consistent of course.

So we could model as:
1. A list of IDs, use the attribute client to translate scope + key (in implicit case) into IDs
2. A map of <scope,key>, use the attribute client to translate IDs (in explicit map case) into scope, key
3. double up the config to hold the scope, key, and attribute ID - seems redundant.
4. Skip the attribute client and use the fact that `id = ${scope}.${key}` to translate for one or the other. I really suggest we don't do that since we've already got the client and this would be breaking encapsulation for no good reason.

Reply:
oh shit! I copied the config and forgot to change the id. What I meant was something like this
```
{
    scope = API
    attributeIds = ["API.lastCalledTimestamp"]
  }
```

Reply:
see above, doesn't give us a way, other than 4, to go from an implicit subdoc path to attribute ID.